foursight_core/templates/header.html

```diff
@@ -0,0 +1,327 @@
+                        {% endif %}
+                    {% else %}
+                        <span style="cursor:pointer;font-weight:bold;color:darkred;" title="Not logged in. Click to login." onclick="create_redir_cookie(); lock.show();">LOGIN</span>
+                    {% endif %}
+                </td>
+            </tr>
+            {% if running_checks != '0' or queued_checks != '0' %}
+                <table width="100%">
+                    <tr style="background:#F9E79F;">
+                        <td style="padding-right:8pt;padding-top:2pt;padding-bottom:2pt;white-space:nowrap;" align="right">
+                            running: <b>{{running_checks}}</b> &nbsp;|&nbsp;
+                            queued: <b>{{queued_checks}}</b>
+                        </td>
+                    </tr>
+                </table>
+            {% endif %}
+        </table>
+    </div>
+    {% if not is_admin %}
+        <div class="container">
+            <div class="boxstyle check-warn" style="margin:20pt;padding:10pt;">
+                Not logged in. Click <u style="cursor:pointer" onclick="create_redir_cookie(); lock.show();"><b>here</b></u> to <span style="cursor:pointer" onclick="create_redir_cookie(); lock.show();">login</span>.
+            </div>
+        </div>
+    {% else %}
+        <div class="container">
+            <div id="content">{% block content %}{% endblock %}</div>
+        </div>
+    {% endif %}
+</div>
+
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.2.4/jquery.min.js"></script>
+<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
+<script src="https://cdn.auth0.com/js/lock/10.24.1/lock.min.js"></script>
+<script type="text/javascript">
+      // set cookie containing the location we were at when we tried to Login
+      // this cookie is re-loaded at /callback and parsed for the url, which
+      // it redirects to upon finishing authentication
+      function create_redir_cookie() {
+        var expr = new Date();
+        expr.setFullYear(expr.getFullYear() + 1);
+        document.cookie = "redir=" + window.location.href + "; path=/; expires=" + expr.toUTCString();
+      }
+</script>
+<script type="text/javascript">
+        var domain = "{{domain}}";
+        var context = "{{context}}";
+        var lock = new Auth0Lock('DPxEwsZRnKDpk0VfVAxrStRKukN14ILB', 'hms-dbmi.auth0.com', {
+            auth: {
+                redirectUrl: 'https://' + domain + context + 'callback/',
+                responseType: 'code',
+                sso: false,
+                params: {scope: 'openid email', prompt: 'select_account'}
+            },
+            socialButtonStyle: 'big',
+            languageDictionary: { title: "Log in" },
+            theme: { logo: "{{favicon}}" },
+            allowedConnections: ['github', 'google-oauth2']
+        });
```

Comment:
This wasn't requested as part of your change, but we need to update this so that it is no longer hard coded. Instead, make an ajax request to the portal's `/auth0_config` endpoint - see https://mastertest.4dnucleome.org/auth0_config?format=json, which should work for all portals and give you (almost) this exact structure.

Reply:
I took care of this; fetching auth0 client ID dynamically.